core/database/foxx/api/user_router.js

```diff
@@ -516,9 +519,47 @@ router.get('/token/set', function(req, res) {
                         refresh: req.queryParams.refresh,
                         expiration: Math.floor(Date.now() / 1000) + req.queryParams.expires_in
                     };
-                    g_db._update(user_id, obj, {
-                        keepNull: false
-                    });
+                    if (token_type && other_token_data) {
+
+                        // NOTE: the edge is only defined for the `u` collection and the `globus_coll` collection.
+                        //  The getUserFromClientID method shows that the user account can also be stored in a UUID collection
+                        //  and in the future support domain accounts. For now, we must ensure that the edge is u to globus_coll
+                        const user_doc = g_db.u.exists(user_id);    // TODO: user doc only necessary for edge key, user_doc._key must not have invalid character '/'
+                                                                    //  which should be kicked out by arango
+                        if (!user_doc) {
+                            throw [g_lib.ERR_INVALID_PARAM, "Error writing to globus_token. No such user in u collection '" + user_id + "'"];
+                        }
+
+                        // find or insert collection
+                        const collection_search_key = other_token_data;
+                        let globus_collection = g_db.globus_coll.exists({_key: collection_search_key});
+                        if (!globus_collection) {
+                            // TODO: what other information should live on the globus_coll document?
+                            globus_collection = g_db.globus_coll.save({_key: collection_search_key, name: "some name we could look up"});
+                        }
+
+                        const token_key = user_doc._key + "_" + globus_collection._key + "_" + token_type;    // TODO: key formatting/convention
+                        const token_doc = {
+                            _key: token_key,
+                            _from: user_id, // the uid field
+                            _to: globus_collection._id,
+                            type: token_type,
+                            ...obj
+                        };
+
+                        console.log("writing to edge ", token_doc);
+                        const token_doc_upsert = g_db.globus_token.insert(
+                            token_doc,
+                            {
+                                overwriteMode: "replace"    // TODO: perhaps use 'update' and specify values for true upsert
```

Comment:
For now we can probably kick this can down the road, but upserts would be helpful in my opinion.